scipy/interpolate/_rgi.py

```diff
@@ -271,6 +271,7 @@ def __init__(self, points, values, method="linear", bounds_error=True,
         elif method in self._SPLINE_METHODS:
             self._validate_grid_dimensions(points, method)
         self.method = method
+        self._spline = None
```

Comment:
Not sure what's going on here?

Reply:
Previously, the RGI implementation replaced the value of `_spline` if it was passed as a custom method in `__call__`, however, this causes an unintended side-effect on that attribute. If the class was declared with a given `_spline`, you shouldn't expect it to change when it gets called with a custom `method` argument, these changes prevent such side effect from taking place.